### PR TITLE
Additional user_settings.h examples

### PR DESCRIPTION
**Pull request by dgarske**

### CODE REVIEW

File: examples/configs/user_settings_template.h [hunk 1]
```diff
@@ -0,0 +1,525 @@
...
+
+#ifndef WOLFSSL_USER_SETTINGS_H
+#define WOLFSSL_USER_SETTINGS_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* If TARGET_EMBEDDED is defined then small target settings are used */
+#if !((defined(__MACH__) || defined(__FreeBSD__) || defined(__linux__)))
```
Comment (lygstate) [inline examples/configs/user_settings_template.h:35]: #if !((defined(__MACH__) || defined(__FreeBSD__) || defined(__linux__) || defined(_WIN32)))

**Comment by lygstate** [comment]
template_all lack
```
   /* for static ciphers */
   #define WOLFSSL_STATIC_RSA
   #define WOLFSSL_STATIC_PSK
   #define WOLFSSL_STATIC_EPHEMERAL
   #define WOLFSSL_SNIFFER
```

**Comment by kaleb-himes** [comment]
Changes are good, just holding off on review due to merge conflicts that will likely arise with https://github.com/wolfSSL/wolfssl/pull/4124 refactor of the examples and addition of the testApps directory (Specifically the include.am file changes in this PR and in 4124)

**Comment by kaleb-himes** [comment]
Moving forward after discussing with @dgarske

**Review by kaleb-himes — APPROVED** [review]
Changes look good, thank you @dgarske, will merge once Jenkins finishes mulling them over.

File: examples/configs/user_settings_all.h [hunk 2]
```diff
@@ -210,6 +211,19 @@ extern "C" {
     #define BOOST_ASIO_USE_WOLFSSL
 #endif
 
+/* TLS static cipher support - off by default */
+#if 0
+    #define WOLFSSL_STATIC_RSA
```
Comment (lygstate) [inline examples/configs/user_settings_all.h:216]: examples/configs/user_settings_template.h doesn't have these

**Pull request merged**